src/components/settings/Recovery/ChooseRecoveryMethodModal.tsx

```diff
@@ -57,11 +61,12 @@ export function ChooseRecoveryMethodModal({ open, onClose }: { open: boolean; on
     },
     mode: 'onChange',
   })
-  const { register, watch } = methods
+  const { register, watch, setValue } = methods
 
   const currentType = watch(FieldNames.recoveryMethod)
 
   const trackOptionChoice = (e: ChangeEvent<HTMLInputElement>) => {
+    isRecoveryMethod(e.target.value) && setValue(FieldNames.recoveryMethod, e.target.value)
```

Comment:
`setValue` should not need to be called as this is inherently called within a registration. Is it otherwise not working?

Reply:
Works with using a `Controller` instead 👍